examples/flow_tracker/flow_tracker.c

```diff
@@ -171,14 +173,16 @@ clear_entries(struct state_info *state_info) {
                 if (update_status(state_info->elapsed_cycles, data) < 0) {
                         return -1;
                 }
-
                 if (!data->is_active) {
                         ret = onvm_ft_remove_key(state_info->ft, key);
                         state_info->num_stored--;
                         if (ret < 0) {
                                 printf("Key should have been removed, but was not\n");
                                 state_info->num_stored++;
                         }
+			printf("++++++++++++++++Send end event msg++++++++++++\n");
+			//_onvm_ft_print_key(key);
+			sending_event_data(FLOW_END_EVENT_ID, 2, (void*)key);
```

Comment:
Is the `2` destination here because that is what you are assuming the PubSub Controller NF's ID will be?  Instead of hard coding `2`, make this a constant like `#define PUBSUB_CONTROLLER_ID 2`

Reply:
(and same thing on line 251)